aten/src/ATen/native/Resize.cpp

```diff
@@ -21,15 +22,29 @@ Tensor& resize_cpu_(Tensor& self, IntArrayRef size) {
 // this function in native_functions.yaml
 Tensor& resize_as_sparse_(Tensor& self, const Tensor& src);
 
-Tensor& resize_as_(Tensor& self, const Tensor& the_template) {
+Tensor& resize_as_(
+    Tensor& self,
+    const Tensor& the_template,
+    c10::optional<MemoryFormat> optional_memory_format) {
   if (self.is_sparse() && the_template.is_sparse()) {
+    TORCH_CHECK(
+        !optional_memory_format.has_value(),
+        "Unsupported memory format for sparse tensor resize_as_ :",
+        optional_memory_format.value_or(MemoryFormat::Contiguous));
     return native::resize_as_sparse_(self, the_template);
   }
+  auto memory_format =
+      optional_memory_format.value_or(MemoryFormat::Contiguous);
   Tensor& result = self.resize_(the_template.sizes());
+  if (memory_format == MemoryFormat::Preserve) {
+    memory_format = the_template.suggest_memory_format();
+  }
+  self.unsafeGetTensorImpl()->empty_tensor_restride(memory_format);
```

Comment:
doesn't this change the behavior, even in cases where there's no channel_last tensors?  i.e. `non_contig.resize_as_(contig.shape)` will make a tensor contiguous when it didn't previously.

Is that what you want?

Reply:
Generally line 38 above `Tensor& result = self.resize_(the_template.sizes());` was already setting up strides to contiguous. 

But thank to this comment I found the only case when it is not intended and might be BC breaking:

```python
x =  torch.empty(2,3).t()
x.resize_as_(x)
# x.is_contiguous() == True but supposed to keep original strides.
```

Gonna fix it now and test cover (we had no such test).